hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/TopicStore.java

```diff
@@ -0,0 +1,144 @@
+                topic.getExpirationTimestamp().toGrpc(),
+                topic.getSequenceNumber(),
+                topic.getRunningHash(),
+                topic.getKey().longValue(),
+                topic.isDeleted());
+    }
+
+    /**
+     * Create a {@link Topic} from a {@link MerkleTopic}
+     * @param topic the {@link MerkleTopic}
+     * @return the {@link Topic}
+     */
+    protected Topic topicFrom(final MerkleTopic topic) {
+        return new TopicImpl(
+                topic.getKey().longValue(),
+                topic.getAdminKey(),
+                topic.getSubmitKey(),
+                topic.getMemo(),
+                topic.getAutoRenewAccountId().num(),
+                topic.getAutoRenewDurationSeconds(),
+                topic.getExpirationTimestamp().getSeconds(),
+                topic.isDeleted(),
+                topic.getSequenceNumber());
+    }
+
+    // TODO : Remove use of TopicMetadata and change to use Topic instead
+    /**
+     * Topic metadata
+     * @param memo topic's memo
+     * @param adminKey topic's admin key
+     * @param submitKey topic's submit key
+     * @param autoRenewDurationSeconds topic's auto-renew duration in seconds
+     * @param autoRenewAccountId topic's auto-renew account id
+     * @param expirationTimestamp topic's expiration timestamp
+     * @param sequenceNumber topic's sequence number
+     * @param runningHash topic's running hash
+     * @param key topic's key
+     * @param isDeleted topic's deleted flag
+     */
+    public record TopicMetadata(
+            Optional<String> memo,
+            Optional<HederaKey> adminKey,
+            Optional<HederaKey> submitKey,
+            long autoRenewDurationSeconds,
+            Optional<Long> autoRenewAccountId,
+            Timestamp expirationTimestamp,
+            long sequenceNumber,
+            byte[] runningHash,
+            long key,
+            boolean isDeleted) {
+        // Overriding equals, hashCode and toString to consider array's content
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            TopicMetadata that = (TopicMetadata) o;
+            return autoRenewDurationSeconds == that.autoRenewDurationSeconds
+                    && sequenceNumber == that.sequenceNumber
+                    && key == that.key
+                    && isDeleted == that.isDeleted
+                    && Objects.equals(memo, that.memo)
+                    && Objects.equals(adminKey, that.adminKey)
+                    && Objects.equals(submitKey, that.submitKey)
+                    && Objects.equals(autoRenewAccountId, that.autoRenewAccountId)
+                    && Objects.equals(expirationTimestamp, that.expirationTimestamp)
+                    && Arrays.equals(runningHash, that.runningHash);
```

Comment:
Are there tests for this `equals()` method (and `hashcode` and `toString`) somewhere?

Reply:
It is on ReadableTopicStoreTest